src/components/Sidebar.js

```diff
@@ -15,16 +24,38 @@ class Sidebar extends Component {
     }
   }
 
+  componentWillReceiveProps (nextProps) {
+    this.setState({
+      items: nextProps.stacks.items
+    })
+  }
+
+  filterStacks (e) {
+    const query = e.currentTarget.value
+    const $items = this.props.stacks.items.filter((item) => {
+      if (item.name.indexOf(query) > -1) {
+        return item.name
+      }
+    })
+
+    this.setState({
+      items: $items
+    })
+  }
+
   render () {
     const routePath = this.props.router.location.pathname
     const showStacks = routePath.indexOf('/stack') >= 0 || routePath === '/new-stack'
-    const { items = [], currentOffset = 0, total = 0 } = this.props.stacks
+    const { currentOffset = 0, total = 0 } = this.props.stacks
 
-    const $stacks = items.map((stack) => {
-      return (
-        <Link key={stack.name} to={`/stacks/${stack.name}`} activeClassName="is-active" className="rka-sidebar-sublink txt-ellipsis">{stack.name}</Link>
-      )
-    })
+    let $stacks
+    if (this.state.items) {
```

Comment:
do you need the `if`? as the constructor also sets a default for `this.state.items` I assume the if is not needed